File: ruby/anagram/anagram.rb
class Anagram
  def initialize(word)
    @word = word.downcase
  end

  def match(list)
    results = []
    list.each do |word|
      results << word if word.downcase.chars.sort == @word.chars.sort
    end
    results.reject { |word| word.downcase == @word }
  end
end

File: javascript/anagram/anagram.js
class Anagram {
  constructor(subject) {
    this.subject = this.clean(subject)
  }

  matches(words) {
    const results = [];
    words.forEach(word => {
      if (this.clean(word) == this.subject) { results.push(word) };
    })

    return results;
  }

  clean(word) {
    return word.split('').sort().join();
  }
}

if (module) module.exports = Anagram;
